### ISSUE TITLE
Markdown parsing is brittle

### ISSUE THREAD

**Issue by ShaneNZ** [conversionfoundry/breeze#26]
Observed on Leftclick site (Breeze version?)

Parsing of the markdown when creating content is somewhat brittle, especially with respect to whitespace. For example:
- a blank line prior to the first item of a list (numbered or bulleted) seems to be required? C
- spaces are required on each line for a numbered list before the list counter, but not for a bulleted list
- strong/italic needs surrounding whitespace; you cannot emphasise part of a word i.e abc__def__ghi does not work
- leading whitespace on a heading line breaks the heading
- placing a markdown list inside `<div class="myclass"> ... </div>` doesn't work (maybe it shouldn't?)

**Comment by isaacfreeman** [comment]
Markdown support is provided by Marquess (https://github.com/fauxparse/marquess), which is a jQuery widget Matt Powell (fauxparse on Github) wrote specifically for the purpose. It supports only a subset of Markdown, and hasn't been updated for a couple of years.

I'd suggest that we switch to some other third-party widget for Markdown support, rather than maintaining Marquess.

**Comment by ShaneNZ** [comment]
Agreed. Had a quick look around, found a couple of options that look OK on
the face of it, left here for future reference:

http://pushingkarma.com/projects/jquery-wysiwym/
http://markitup.jaysalvat.com/examples/markdown/
http://ajaxorg.github.com/ace/build/kitchen-sink.html
http://blog.aisleten.com/2008/09/11/rails-textile-and-javascript-wysiwyg-roundup/(comparison
article for different WYSIWYM editors, textile focused but
includes (a bias against) markdown :)

Some of these support Textile as well - I have no preference, and
transitioning would be a pain for existing sites, but there might be
tech/strategic reasons why it makes sense (haven't thought about that).
